### PR TITLE
ROU-3174: Auto zoom for polyline shape

### PR DESCRIPTION
**Pull request by OS-giulianasilva**
This PR is for adding the new Optional Config AutoZoomOnShapes.

### What was done
* Add the options that allow AutoZoom to adjust to fit all shapes and markers displayed on the map.
   * Add the autoZoomOnShapes to the Map configuration.
   * Add the providerBounds getter to each shape.
   * Updated the Zoom.refreshZoom() to consider the shapes' bounds.
* Note: For Leaflet, this feature does not update dynamically.

### Test Steps
1. Go to a page with a Map that contains Shapes
2. Set the Zoom = AutoZoom and AutoZoomOnShapes = true
3. Check that the Map Zoom automatically fits all the Shapes in the Map

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: src/OSFramework/Maps/Configuration/IConfigurationMap.ts [hunk 1]
```diff
@@ -11,5 +11,6 @@ namespace OSFramework.Maps.Configuration {
         type?: Enum.OSMap.Type;
         uniqueId: string;
         zoom: Enum.OSMap.Zoom;
+        autoZoomOnShapes?: boolean;
```
Comment (rugoncalves) [inline src/OSFramework/Maps/Configuration/IConfigurationMap.ts:14]: alphabetic order - fail! :)

File: src/Providers/Maps/Google/Configuration/OSMap/GoogleMapConfig.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ namespace Provider.Maps.Google.Configuration.OSMap {
         public type: OSFramework.Maps.Enum.OSMap.Type;
         public uniqueId: string;
         public zoom: OSFramework.Maps.Enum.OSMap.Zoom;
+        public autoZoomOnShapes: boolean;
```
Comment (rugoncalves) [inline src/Providers/Maps/Google/Configuration/OSMap/GoogleMapConfig.ts:18]: alphabetic order - fail! :)

File: src/Providers/Maps/Leaflet/Configuration/OSMap/LeafletMapConfiguration.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ namespace Provider.Maps.Leaflet.Configuration.OSMap {
         public offset: OSFramework.Maps.OSStructures.OSMap.Offset;
         public uniqueId: string;
         public zoom: OSFramework.Maps.Enum.OSMap.Zoom;
+        public autoZoomOnShapes: boolean;
```
Comment (rugoncalves) [inline src/Providers/Maps/Leaflet/Configuration/OSMap/LeafletMapConfiguration.ts:12]: alphabetic order - fail! :)

File: src/Providers/Maps/Leaflet/Configuration/Shape/BasicShapeConfig.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ namespace Provider.Maps.Leaflet.Configuration.Shape {
         public strokeColor: string;
         public strokeOpacity: number;
         public strokeWeight: number;
+        public autoZoomOnShape: boolean;
```
Comment (rugoncalves) [inline src/Providers/Maps/Leaflet/Configuration/Shape/BasicShapeConfig.ts:15]: alphabetic order - fail! :)

File: src/Providers/Maps/Leaflet/Features/Zoom.ts [hunk 5]
```diff
@@ -40,24 +63,16 @@ namespace Provider.Maps.Leaflet.Feature {
 
         public refreshZoom(): void {
             if (this._map.features.zoom.isAutofit) {
```
Comment (rugoncalves) [inline src/Providers/Maps/Leaflet/Features/Zoom.ts:65]: This method can now maybe be moved to the abstractMap class. What do you think @OS-giulianasilva?

**Review by rugoncalves — APPROVED** [review]

File: src/Providers/Maps/Leaflet/Features/Zoom.ts (hunk 5 above)
Comment (rugoncalves) [reply src/Providers/Maps/Leaflet/Features/Zoom.ts:65]: Nevermind!

**Review by rugoncalves — DISMISSED** [review]

File: src/Providers/Maps/Google/Features/Zoom.ts [hunk 6]
```diff
@@ -30,6 +30,29 @@ namespace Provider.Maps.Google.Feature {
             );
         }
 
+        private _setBounds(useShapes: boolean) {
+            const bounds = new google.maps.LatLngBounds();
+            this._map.markers.forEach(function (item) {
+                if (item.provider === undefined) return;
+                const loc = item.provider.position.toJSON();
```
Comment (BenOsodrac) [inline src/Providers/Maps/Google/Features/Zoom.ts:37]: I think we could remove the const and use the method directly

File: src/Providers/Maps/Google/Features/Zoom.ts [hunk 7]
```diff
@@ -30,6 +30,29 @@ namespace Provider.Maps.Google.Feature {
...
+                bounds.extend(loc);
+            });
+
+            if (useShapes) {
+                this._map.shapes.forEach(function (item) {
+                    if (item.provider === undefined) return;
+                    const loc = item.providerBounds;
```
Comment (BenOsodrac) [inline src/Providers/Maps/Google/Features/Zoom.ts:44]: same

**Review by BenOsodrac — APPROVED** [review]

File: src/Providers/Maps/Google/Features/Zoom.ts (hunk 6 above)
Comment (OS-giulianasilva) [reply src/Providers/Maps/Google/Features/Zoom.ts]: Fixed!

File: src/Providers/Maps/Google/Features/Zoom.ts (hunk 7 above)
Comment (OS-giulianasilva) [reply src/Providers/Maps/Google/Features/Zoom.ts]: Fixed!

**Review by BenOsodrac — APPROVED** [review]

**Pull request merged**